scripts/codegen.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
```

Comment:
Should this be named something that makes it clearer that it generates from the IDL?

Reply:
I added a text blurb in main (gets displayed when using `--help`). Not sure about other names, let me know of suggestions. Generally 'codegen' seemed close enough to think of code generation, however if better naming exist the renames should be easy.